bndtools.core/src/bndtools/wizards/project/ImportBndWorkspaceWizard.java

```diff
@@ -0,0 +1,451 @@
+            IPackageFragmentRoot root = javaProject.getPackageFragmentRoot(source);
+            List<IClasspathEntry> entries = new ArrayList<>(Arrays.asList(javaProject.getRawClasspath()));
+            entries.add(JavaCore.newSourceEntry(root.getPath(), null, defaultOutputLocation ? null : outputRoot.getPath()));
+            javaProject.setRawClasspath(entries.toArray(new IClasspathEntry[entries.size()]), monitor);
+            createFolderIfNecessary(source, monitor);
+        }
+        // Test-Source
+        javaProject.setOutputLocation(sourceOutput.getFullPath(), null);
+        if (!bndProject.getSrcOutput().equals(bndProject.getTestOutput())) {
+            IFolder testOutput = workspaceProject.getFolder(URIUtil.toPath(bndProject.getTestOutput().toURI()).makeRelativeTo(workspaceProject.getLocation()));
+            IPackageFragmentRoot testOutputRoot = javaProject.getPackageFragmentRoot(testOutput);
+            IFolder testSource = workspaceProject.getFolder(URIUtil.toPath(bndProject.getTestSrc().toURI()).makeRelativeTo(workspaceProject.getLocation()));
+            // Now the created source folder should be added to the class entries of the project, otherwise compilation will fail
+            IPackageFragmentRoot root = javaProject.getPackageFragmentRoot(testSource);
+            List<IClasspathEntry> entries = new ArrayList<>(Arrays.asList(javaProject.getRawClasspath()));
+            entries.add(JavaCore.newSourceEntry(root.getPath(), null, testOutputRoot.getPath()));
+            javaProject.setRawClasspath(entries.toArray(new IClasspathEntry[entries.size()]), monitor);
+            createFolderIfNecessary(testSource, monitor);
+        }
+
+        // Generated Artifact
+        IFolder generated = workspaceProject.getFolder(URIUtil.toPath(bndProject.getTarget().toURI()).makeRelativeTo(workspaceProject.getLocation()));
+        createFolderIfNecessary(generated, monitor);
+    }
+
+    private void createFolderIfNecessary(IFolder folder, IProgressMonitor monitor) throws CoreException {
+        if (!folder.exists()) {
+            folder.create(true, true, monitor);
+        }
+    }
+
+    /**
+     * The Java-Nature doesn't add a JRE-Container, so we add one
+     *
+     * @param javaProject
+     *            the Java project which should get enhanced with LibraryContainer
+     * @param javacTarget
+     * @param monitor
+     *            current IProgressMonitor
+     * @throws JavaModelException
+     */
+    private void addSystemLibraryContainer(IJavaProject javaProject, String javacTarget, IProgressMonitor monitor) throws JavaModelException {
+        // TODO resolution of ExecutionEnvironment sketchy. Which Bnd property should be used? It seems theres a BncConstant missing for setting an EE globally
+        IExecutionEnvironment environment = JavaRuntime.getExecutionEnvironmentsManager().getEnvironment("JavaSE-" + javacTarget);
```

Comment:
Lets just check the BndConstants.JAVAC_SOURCE or BndConstants.JAVAC_TARGET property if they exist.